zaza/openstack/utilities/openstack.py

```diff
@@ -2332,6 +2332,12 @@ def verify(stdin, stdout, stderr):
                 password=password, privkey=privkey, verify=verify)
 
 
+# ssh.connect() calls read_all() which can raise an EOFError, see
+# * https://docs.paramiko.org/en/stable/api/packet.html
+# * https://github.com/paramiko/paramiko/issues/925
+# Using tenacity to retry a few times.
+@tenacity.retry(wait=tenacity.wait_exponential(multiplier=1, min=2, max=10),
+                reraise=True, stop=tenacity.stop_after_attempt(3))
```

Comment:
I'd prefer to add a  `retry`  boolean option to `ssh_command` to control whether to do the retry, happy for this to default to`True`.